### ISSUE TITLE
the network output will become nan

### ISSUE THREAD

**Issue by 15066952332** [RonyAbecidan/ManTraNet-pytorch#13]
![image](https://user-images.githubusercontent.com/60381409/215236395-7d63097f-d043-4d95-9058-3218ac8bae6c.png)
![image](https://user-images.githubusercontent.com/60381409/215236487-d537d9ac-0359-44e1-aeff-5d64de464726.png)
I am very grateful to the author for the pytorch version code. During my training, I found that the network output will become nan. Has the author encountered such a problem?

**Comment by RonyAbecidan** [comment]
Hello, 

Normally, for each image, the expected output should be a binary map with just pixels equal to 0 or 1 but I guess that you have some masks that are not satisfying this constraint. This may explain your error ;)

**Comment by 15066952332** [comment]
Thank you very much for your answer, but I encountered this problem during the training process. During the training, the output of the first few iterations is 0~1, and then the iterations will appear nan. Have you ever encountered this kind of problem? Or is the size of my data set image related to this? My picture input is 512*512，我发现模型是从IMTFE部分输出变成的nan
![image](https://user-images.githubusercontent.com/60381409/215384451-c508cbfb-0424-4b92-82f5-d152e92e33e0.png)

**Comment by RonyAbecidan** [comment]
I am not sure why it is becoming nan suddenly. Can you share your training code ? I will check the IMTFE meanwhile

**Comment by 15066952332** [comment]
Thanks a lot，This is the code of my train
![image](https://user-images.githubusercontent.com/60381409/215702466-7aaba1b0-4e94-4008-9bcd-1d0d3d25e174.png)
The model is defined as Mantranet
“Can you also check that all your target masks contain only 0 or 1 pixels ?”Some of my target masks are tampered images (the mask has 0 and 1), and some are untampered images (the mask is all 0). Will this data set be affected?

**Comment by RonyAbecidan** [comment]
I have some comments about your code that may explain your error :

1 - You need to provide probabilities to BCE_loss in "seg_out". To do so, you have to use "nn.Sigmoid()" and apply it to seg_out before passing it in the BCE_loss.
2 - I don't understand why do you need to use "nn.functional.interpolate". Mantranet is supposed to give you back an output with the same size as the input.

**Comment by RonyAbecidan** [comment]
Hello, 

Did you fix your issues finally ?
